src/scaffoldfitter/fitter.py

```diff
@@ -1255,8 +1257,21 @@ def calculateGroupDataProjections(self, fieldcache, group, dataGroup, meshGroup,
                     result = meshLocation.assignMeshLocation(fieldcache, element, xi)
                     assert result == RESULT_OK, \
                         "Error: Failed to assign data projection mesh location for group " + groupName
-                    dataProjectionNodesetGroup.addNode(node)
+                    result, projectionLength = self._dataErrorField.evaluateReal(fieldcache, 1)
+                    if projectionLength > maximumProjectionLength:
+                        maximumProjectionLength = projectionLength
+                    if outlierLength < 0.0:
+                        # store and filter once we know the maximum
+                        dataProjectionLengths.append((node.getIdentifier(), projectionLength))
+                    if (outlierLength <= 0.0) or (projectionLength <= outlierLength):
+                        dataProjectionNodesetGroup.addNode(node)
             node = nodeIter.next()
+        if outlierLength < 0.0:
+            relativeOutlierLength = (1.0 + outlierLength) * maximumProjectionLength
+            for nodeIdentifier, projectionLength in dataProjectionLengths:
+                if projectionLength > relativeOutlierLength:
+                    node = dataGroup.findNodeByIdentifier(nodeIdentifier)
+                    dataProjectionNodesetGroup.removeNode(node)
```

Comment:
Is it a performance issue to lookup the node rather than store the node locally? This could be refactored to use dictionaries and comprehensions to make things slightly quicker.

Reply:
I never store nodes locally as they could be implemented as a flyweight model; there may not be an underlying object, only a temporary.
However, to make this more efficient I can make a temporary conditional field and remove nodes conditional, if efficiency is what you want.